app/src/main/java/com/support/android/designlibdemo/CheeseDetailActivity.java

```diff
@@ -42,6 +43,17 @@ public void onCreate(Bundle savedInstanceState) {
         final String cheeseName = intent.getStringExtra(EXTRA_NAME);
 
         final Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+        if(Build.VERSION.SDK_INT>=21){
+            int result = 0;
+            int resourceId = getResources().getIdentifier(
+                    "status_bar_height", "dimen", "android");
+            if (resourceId > 0) {
+                result = getResources().getDimensionPixelSize(resourceId);
```

Comment:
I found that you are right on the problem of android version , but I still
don't understand why the result should be double.I run what I wrote in my
phone(Xiaomi aries android 5.0.2), and the position of Toolbar seems no
problem ......

2015-06-09 18:12 GMT+08:00 Steffan Davies <notifications@github.com>:

> In
> app/src/main/java/com/support/android/designlibdemo/CheeseDetailActivity.java
> <https://github.com/chrisbanes/cheesesquare/pull/12#discussion_r31998895>:
>
> > @@ -42,6 +43,17 @@ public void onCreate(Bundle savedInstanceState) {
> >          final String cheeseName = intent.getStringExtra(EXTRA_NAME);
> >
> >          final Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
> > +        if(Build.VERSION.SDK_INT>=21){
> > +            int result = 0;
> > +            int resourceId = getResources().getIdentifier(
> > +                    "status_bar_height", "dimen", "android");
> > +            if (resourceId > 0) {
> > +                result = getResources().getDimensionPixelSize(resourceId);
>
> Sorry, I should have explained this better. If you make the Build.VERSION.SDK_INT
> == 21 alteration above, you also need to remove
> android:fitsSystemWindows="true" below as it causes the Toolbar to be
> positioned incorrectly on API level 22. Once those changes are made, you
> also need to change the above line to result =
> getResources().getDimensionPixelSize(resourceId) * 2; to position the
> Toolbar correctly on API level 21.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/chrisbanes/cheesesquare/pull/12/files#r31998895>.
>